File: Deal/src/main/java/ru/leonov/deal/client/ConveyorAppClient.java
package ru.leonov.deal.client;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.PostMapping;
import ru.leonov.deal.dto.CreditDTO;
import ru.leonov.deal.dto.LoanApplicationRequestDTO;
import ru.leonov.deal.dto.LoanOfferDTO;
import ru.leonov.deal.dto.ScoringDataDTO;

import java.util.List;

@FeignClient(name = "conveyor-app-client", url = "${conveyor-app.hostname}")
public interface ConveyorAppClient {

    @PostMapping(value = "/conveyor/offers")
    List<LoanOfferDTO> requestLoanOffer(LoanApplicationRequestDTO requestDTO);

    @PostMapping(value = "/conveyor/calculation")
    CreditDTO requestCreditCalculation(ScoringDataDTO scoringDataDTO);

}

File: Deal/deal_impl/src/main/kotlin/ru/leonov/deal/client/ConveyorAppClient.kt
package ru.leonov.deal.client

import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.PostMapping
import ru.leonov.deal.dto.CreditDTO
import ru.leonov.deal.dto.LoanApplicationRequestDTO
import ru.leonov.deal.dto.LoanOfferDTO
import ru.leonov.deal.dto.ScoringDataDTO

@FeignClient(name = "conveyor-app-client", url = "\${conveyor-app.hostname}")
interface ConveyorAppClient {

    @PostMapping(value = ["/conveyor/offers"])
    fun requestLoanOffer(requestDTO: LoanApplicationRequestDTO): MutableList<LoanOfferDTO>?

    @PostMapping(value = ["/conveyor/calculation"])
    fun requestCreditCalculation(scoringDataDTO: ScoringDataDTO): CreditDTO?
}
